pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -722,3 +723,64 @@ def partition(low, high, pick_pivot_element):
         array._modify(force=True)
 
     return array
+
+def longest_common_subsequence(seq1, seq2) -> tuple:
+    """
+    Implements Longest Common Subsequence
+
+    Parameters
+    ========
+
+    seq1: Any 1D data structure that can be indexed (like list, tuple, string)
+    seq2: Any 1D data structure that can be indexed (like list, tuple, string)
+
+    Returns
+    =======
+
+    output: tuple
+        The first element of the tuple represents the length of longest common subsequence and
+        the second element is the longest common subsequence itself.
+        Common subsequence will be of the same data type as that of input sequences.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import longest_common_subsequence as LCS
+    >>> LCS("ABCDEF", "ABBCDDDE")
+    (5, 'ABCDE')
+    >>> arr1 = ['A', 'P', 'P']
+    >>> arr2 = ['A', 'p', 'P', 'S', 'P']
+    >>> LCS(arr1, arr2)
+    (3, ['A', 'P', 'P'])
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+    """
+    row, col = len(seq1), len(seq2)
+    check_mat = {0:[0 for _ in range(col+1)]}
+    for i in range(row):
+        check_mat[i+1]=[0 for _ in range(col+1)]
+        for j in range(col):
+            if (seq1[i] == seq2[j]):
+                check_mat[i+1][j+1] = check_mat[i][j]+1
+            else:
+                check_mat[i+1][j+1] = max(check_mat[i+1][j], check_mat[i][j+1])
+
+    lcseq, lclen = [], check_mat[row][col]
+    while(row > 0  and col > 0):
+        if(check_mat[row][col] == check_mat[row][col-1]):
+            col -= 1
+        elif(check_mat[row][col] == check_mat[row-1][col]):
+            row -= 1
+        else:
+            lcseq.append(seq1[row-1])
+            row -= 1
+            col -= 1
+
+    if(type(seq1) == str):
+        lcseq = ''.join(lcseq)
+    if(type(seq1) == tuple):
+        lcseq = tuple(lcseq)
+    return (lclen, lcseq[::-1])
```

Comment:
This part is cryptic. We should keep the input types restricted to `OneDimensionalArray` only, otherwise such things will create problems while porting the code to statically typed languages like C++.
In addition, applying longest common subseqeunce on strings would be confusing because there is already something called longest common substring.
Hence the final API should be, accept two `OneDimensionalArray` objects and return a `OneDimensionalArray`. 

P.S. That is why doing some background lookups are preferred for discussing APIs rather than just directly coding out things and keep changing frequently.

Reply:
Yeah next time onwards we discuss and start the implementation and I will try to implement the above in One dimensional